### PR TITLE
CDSObject type to enum

### PR DESCRIPTION
**Pull request by whyman**

### CODE REVIEW

File: src/cds_objects.cc [hunk 1]
```diff
@@ -171,7 +163,7 @@ void CdsItem::validate() const
 
 CdsItemExternalURL::CdsItemExternalURL()
 {
-    objectType |= OBJECT_TYPE_ITEM_EXTERNAL_URL;
+    objectType = Type::EXTERNAL_URL;
```
Comment (KarlStraussberger) [inline src/cds_objects.cc:166]: `::EXTERNAL_URL `: You need to set Type::ITEM | Type::EXTERNAL_URL otherwise the search for items in sql_database does not return them anymore.

Also check the usages of `isCdsItem` and `isCdsPureItem`

File: src/database/sql_database.cc [hunk 2]
```diff
@@ -938,14 +938,14 @@ std::vector<std::shared_ptr<CdsObject>> SQLDatabase::browse(BrowseParam& param)
             auto zero = std::string("0 = 1");
             where.push_back(std::move(zero));
         } else if (getContainers && !getItems) {
-            where.push_back(fmt::format("{} = {:d}", browseColumnMapper->mapQuoted(BrowseCol::ObjectType), OBJECT_TYPE_CONTAINER));
+            where.push_back(fmt::format("{} = {:d}", browseColumnMapper->mapQuoted(BrowseCol::ObjectType), CdsObject::Type::CONTAINER));
             orderBy = fmt::format(" ORDER BY {}", orderByCode());
         } else if (!getContainers && getItems) {
-            where.push_back(fmt::format("({0} & {1}) = {1}", browseColumnMapper->mapQuoted(BrowseCol::ObjectType), OBJECT_TYPE_ITEM));
+            where.push_back(fmt::format("({0} & {1}) = {1}", browseColumnMapper->mapQuoted(BrowseCol::ObjectType), CdsObject::Type::ITEM));
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:944]: `::EXTERNAL_URL ` See comment above

File: src/database/sql_database.cc [hunk 3]
```diff
@@ -1108,9 +1108,9 @@ int SQLDatabase::getChildCount(int contId, bool containers, bool items, bool hid
         fmt::format("{} = {}", identifier("parent_id"), contId)
     };
     if (containers && !items)
-        where.push_back(fmt::format("{} = {}", identifier("object_type"), OBJECT_TYPE_CONTAINER));
+        where.push_back(fmt::format("{} = {}", identifier("object_type"), CdsObject::Type::CONTAINER));
     else if (items && !containers)
-        where.push_back(fmt::format("({0} & {1}) = {1}", identifier("object_type"), OBJECT_TYPE_ITEM));
+        where.push_back(fmt::format("({0} & {1}) = {1}", identifier("object_type"), CdsObject::Type::ITEM));
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:1113]: `::EXTERNAL_URL ` See comment above

File: src/database/sql_database.cc [hunk 4]
```diff
@@ -1141,9 +1141,9 @@ std::map<int, int> SQLDatabase::getChildCounts(const std::vector<int>& contId, b
         fmt::format("{} IN ({})", identifier("parent_id"), fmt::join(contId, ","))
     };
     if (containers && !items)
-        where.push_back(fmt::format("{} = {}", identifier("object_type"), OBJECT_TYPE_CONTAINER));
+        where.push_back(fmt::format("{} = {}", identifier("object_type"), CdsObject::Type::CONTAINER));
     else if (items && !containers)
-        where.push_back(fmt::format("({0} & {1}) = {1}", identifier("object_type"), OBJECT_TYPE_ITEM));
+        where.push_back(fmt::format("({0} & {1}) = {1}", identifier("object_type"), CdsObject::Type::ITEM));
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:1146]: `::EXTERNAL_URL ` See comment above

**Comment by whyman** [comment]
I wrote out a whole thing but github didnt commit it.

TLDR closing this to tidy up the existing OO tree first.
